File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/entity/Booking.java
package at.ac.tuwien.sepm.groupphase.backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.FetchType;
import javax.persistence.CascadeType;
import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "bookings")
public class Booking {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long id;

    @Column(nullable = false)
    private Boolean isPaid;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, mappedBy = "booking")
    private List<Ticket> tickets;

    @Column(nullable = false)
    private LocalDate date;

    @Column
    private Boolean isCanceled;

    @ToString.Exclude
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    private ApplicationUser user;
}

File: frontend/src/app/dtos/event/booking.ts
import {TicketToCreate} from './ticket';

export class BookingToCreate {
  id: number;
  isPaid: boolean;
  tickets: TicketToCreate[];
  date: Date;
  userId: number;
  isCanceled: boolean;
  constructor(id: number, isPaid: boolean, tickets: TicketToCreate[], date: Date, userId: number, isCanceled: boolean) {
    this.id = id;
    this.isPaid = isPaid;
    this.tickets = tickets;
    this.date = date;
    this.userId = userId;
    this.isCanceled = isCanceled;
  }
}
